runtime/vm/CRIUHelpers.cpp

```diff
@@ -675,6 +666,15 @@ runInternalJVMRestoreHooks(J9VMThread *currentThread, const char **nlsMsgFormat)
 	}
 #endif /* JAVA_SPEC_VERSION >= 20 */
 
+	if (J9_ARE_ALL_BITS_SET(vm->checkpointState.flags, J9VM_CRIU_IS_NON_PORTABLE_RESTORE_MODE)) {
+		PORT_ACCESS_FROM_JAVAVM(vm);
+		vm->portLibrary->isCheckPointAllowed = FALSE;
+		vm->checkpointState.flags &= ~J9VM_CRIU_IS_CHECKPOINT_ALLOWED;
+		j9port_control(J9PORT_CTLDATA_CRIU_SUPPORT_FLAGS, OMRPORT_CRIU_SUPPORT_ENABLED | J9OMRPORT_CRIU_SUPPORT_FINAL_RESTORE);
+	}
```

Comment:
Assuming we want to change `isCheckPointAllowed` value early, should this be moved even earlier such as the beginning of `criuCheckpointJVMImpl()`?

Reply:
It just has to be done before any Java code runs.